src/Resources/skeleton/crud/controller/Controller.tpl.php

```diff
@@ -43,10 +47,9 @@ public function new(Request $request): Response
     {
         $<?= $entity_var_singular ?> = new <?= $entity_class_name ?>();
         $form = $this->createForm(<?= $form_class_name ?>::class, $<?= $entity_var_singular ?>);
-        $form->handleRequest($request);
 
-        if ($form->isSubmitted() && $form->isValid()) {
-            $em = $this->getDoctrine()->getManager();
+        if ($form->handleRequest($request)->isSubmitted() && $form->isValid()) {
+            $em = $this->get('doctrine.orm.entity_manager');
```

Comment:
This works only because the service is public? If yes, then I think that's not a good idea, we should encourage ppl to not rely on any public services, so that they learn it's better to declare your dependencies.

Reply:
It relies one the same behaviour as the previous code. It just saves some calls.
I might agree with you anyway . Could you open an issue about it ?
Btw it's the same for the form factory

Reply:
In the docs, we've so far decided to never use `$this->get()` anymore, and instead to:

1) Use a shortcut method if it's available (e.g. `$this->getDoctrine()`)
2) if there is no shortcut, use action DI

And I think that is the right thing to do. So I think we should not make this change. And if we *do* want to make this change, let's discuss on symfony-docs - so we can keep the generated code consistent with the docs :)